src/main/scala/uk/gov/hmrc/domain/nino.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.domain
+
+import play.api.libs.json.{Reads, Writes}
+
+import scala.util.Try
+
+sealed trait ShortOrSuffixedNino extends TaxIdentifier with SimpleName {
```

Comment:
On one hand - it might be useful to implement isValid in ShortOrSuffixedNino, on the other hand – we have parse(nino:String) method, do we need both isValid and parse methods publicly available?